com.unity.ml-agents/Runtime/Sensors/BufferSensorComponent.cs

```diff
@@ -9,6 +9,19 @@ namespace Unity.MLAgents.Sensors
     [AddComponentMenu("ML Agents/Buffer Sensor", (int)MenuGroup.Sensors)]
     public class BufferSensorComponent : SensorComponent
     {
+
+        /// <summary>
+        /// Name of the generated <see cref="bufferSensor"/> object.
+        /// Note that changing this at runtime does not affect how the Agent sorts the sensors.
+        /// </summary>
+        public string SensorName
```

Comment:
Actually, I think all of these (name, ObservableSize, MaxNumObservables) need 
* private storage
* public get/set accessors.
* a custom editor (don't let any of the fields get updated during play mode)

Reply:
Ok, I just found the `CameraSensorComponentEditor` and I will use that as a base.

Reply:
Sounds good. I think this might even be simpler than CameraSensorComponentEditor, since I don't think there's anything that reasonably editable at runtime.

Reply:
done